thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionReplayOnboardingTask.java

```diff
@@ -44,6 +53,23 @@ public FunctionReplayOnboardingTask() {
    */
   @Override
   public void run() {
+
+    try {
+      Response response = initDetectionJob();
+      Map<Long, Long> functionIdToJobIdMap = (Map<Long, Long>) response.getEntity();
+      for (long jobId : functionIdToJobIdMap.values()) {
+        JobStatus jobStatus = detectionJobScheduler.waitForJobDone(jobId);
+        if (JobStatus.FAILED.equals(jobStatus) || JobStatus.TIMEOUT.equals(jobStatus)) {
+          throw new InterruptedException("Get Job Status: " + jobStatus);
+        }
+      }
+
+    } catch (Exception e) {
+      throw new UnsupportedOperationException(e);
```

Comment:
I don't think we should throw UnsupportedOperationException, which implies the method, run(), is not supported or implemented. You could throw IllegalStateException or rethrow the exception e.